### PR TITLE
Replicate physician registration page closes #88

### LINKED ISSUE #88: URGENT: Replicate Physician Registration Page with Unbounce Implementation
The new Unbounce landing page will be using the URL the existing Physician Registration page. We will need to replicate the registration fields on the Sign In page on the right hand side of the screen.
 
**Steps:**
1. User will select "Join the Community"
2. User will complete fields below on the right hand side of screen. When "Request Invite Now" is selected, user will be presented with "Registration Sent" popup.
(**URL**: https://www.covid19webapp.com/bIiOOIIqgwEXwUU3SaD0F9)

**Fields:**
- Email
- Role: (Dropdown)
     - Physician
     -Healthcare Provider
     -Other

-Select Country (Dropdown)

-Profile Link (Clinic)

-Medical Provider Number (eg. NPI Identifier)

-Be anonymous (checkbox)

**New Landing Page**
image

**Current Sign/In Register**
image

**Existing Registration Page**
image

**Registration Sent Popup**
image

┆Issue is synchronized with this [Trello card](https://trello.com/c/QTIf6oJ6) by [Unito](https://www.unito.io/learn-more)

### PR DESCRIPTION
**Pull request by mrememisaac**
## Motivation

1. Use redirect rather than popup when authenticating users
2. Eliminate the tedium of importing users from unbounce, by creating users on our active directory 
3. Use branded signup and login pages to help users feel at home
4. Enable physicians provide additional information at registration

## Description
Uses the redirect flow instead of popup auth flow

Fixes # (88)

## Implementation

The function isIE() now returns true always. That's how we ensure we are always redirecting.
We also set resources for the openid and profile graph scopes

## Type of change
- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature

## Screenshots/Links
image

image

### CODE REVIEW

File: client/public/home.css [hunk 1]
```diff
@@ -0,0 +1,402 @@
+* { box-sizing: border-box; }
+
+html, body{
+	height:100vh;
+	padding:0;
+	margin:0;
```
Comment (gitjul) [inline client/public/home.css:6]: pleas please please use 2 spaces for indentation. `prettier` can take care of that for you, or maybe it's not configured for css in the project? if it's not then it definitely should be, meanwhile you could install it locally and run when needed

File: client/public/home.css [hunk 2]
```diff
@@ -0,0 +1,402 @@
...
+}
+
+.heroSection{
+	color: rgba(0,0,0,.87);
+	/* background: -webkit-gradient(linear,left top,left bottom,from(rgba(71,105,123,.7)),to(rgba(71,105,123,.7))), url(https://ememisaac2root.blob.core.windows.net/sites/covid19/login-bg.bddb78cd.jpg) no-repeat center center fixed; 
+	background: linear-gradient(rgba(71, 105, 123, 0.7), rgba(71, 105, 123, 0.7)) url(https://ememisaac2root.blob.core.windows.net/sites/covid19/login-bg.bddb78cd.jpg) no-repeat center center fixed; 
+    -webkit-background-size: cover;
+    -moz-background-size: cover;
+    -o-background-size: cover;
+    background-size: cover; */
```
Comment (gitjul) [inline client/public/home.css:39]: please remove commented code - unless there is a good reason not to do that?

File: client/public/home.css [hunk 3]
```diff
@@ -0,0 +1,402 @@
...
+    color: #fff;
+    background: #006eaa;
+    box-shadow: 0 0 0 1px transparent inset;
+	cursor: pointer;
+    display: inline-block;
+    min-height: 1em;
+    outline: 0;
+    border: none;
+    vertical-align: baseline;
+	font-family: Lato,'Helvetica Neue',Arial,Helvetica,sans-serif;
```
Comment (gitjul) [inline client/public/home.css:132]: inconsistent indentation. again - `prettier` needed. why worry about things like this ourselves

File: client/src/AuthProvider.js [hunk 4]
```diff
@@ -27,8 +27,34 @@ export default (C) =>
       };
     }
 
+    async handleRedirectCallback(tokenReceivedCallback, errorReceivedCallback){
+      
+    }
```
Comment (gitjul) [inline client/src/AuthProvider.js:32]: what is this for?

File: client/src/AuthProvider.js [hunk 5]
```diff
@@ -27,8 +27,34 @@ export default (C) =>
...
+      // }
+      // const account = msalApp.getAccount();
+      // const resp = msalApp.acquireTokenRedirect(request, "http://localhost:3000")
+      // .then(data => {
+      //   console.log(data);
+      // })
+      // .catch((error) => {
+      //   console.log(error);
+      // });
+      // return resp;
```
Comment (gitjul) [inline client/src/AuthProvider.js:57]: please remove commented code unless there's a good reason not to

File: client/public/home.html [hunk 6]
```diff
@@ -0,0 +1,30 @@
+<!DOCTYPE html>
```
Comment (gitjul) [inline client/public/home.html:1]: indentation in this whole file badly needs to be fixed. `prettier`? :)

File: client/public/home.css [hunk 7]
```diff
@@ -0,0 +1,402 @@
...
+
+#continue{
+    font-size: 18px;
+    font-weight: 400;
+    height: auto;
+    color: #fff;
+    background: #006eaa;
+    box-shadow: 0 0 0 1px transparent inset;
+	cursor: pointer;
+    /* display: inline-block; */
```
Comment (gitjul) [inline client/public/home.css:321]: we shouldn't keep commented code as a rule

File: client/public/home.css [hunk 8]
```diff
@@ -0,0 +1,402 @@
...
+    min-height: 1em;
+    outline: 0;
+    border: none;
+    vertical-align: baseline;
+	font-family: Lato,'Helvetica Neue',Arial,Helvetica,sans-serif;
+    /* margin: 0 .25em 0 0; */
```
Comment (gitjul) [inline client/public/home.css:327]: we shouldn't keep commented code as a rule

File: client/public/home.css [hunk 9]
```diff
@@ -0,0 +1,402 @@
...
+    padding: .78571429em 1.5em .78571429em;
+    text-transform: none;
+    text-shadow: none;
+	line-height: 1em;
+    font-style: normal;
+    text-align: center;
+    text-decoration: none;
+    border-radius: .28571429rem;
+	/* grid-column-start:2;
+	grid-column-end: end; */
```
Comment (gitjul) [inline client/public/home.css:337]: we shouldn't keep commented code as a rule

File: client/public/home.css [hunk 10]
```diff
@@ -0,0 +1,402 @@
...
+}
+
+.helpLink {
+	display:none;
+}
+
+.dropdown_single {
+	width: 100%;
+	padding:10px;	
+	/* margin-bottom: 1.2em; */
```
Comment (gitjul) [inline client/public/home.css:352]: we shouldn't keep commented code as a rule

File: client/public/home.css [hunk 11]
```diff
@@ -0,0 +1,402 @@
...
+	display: grid;
+    grid-template-columns: 2fr 1fr 1fr 3fr;
+	align-items: center;
+	column-gap: 10px;
+}
+
+#cancel{
+	display: none;
+	/* grid-column-start: 1;
+	grid-column-end: 2; */
```
Comment (gitjul) [inline client/public/home.css:369]: we shouldn't keep commented code as a rule

File: client/public/home.css [hunk 12]
```diff
@@ -0,0 +1,402 @@
...
+
+#email_ver_but_edit{
+	font-size: 18px;
+    font-weight: 400;
+    height: auto;
+    color: #fff;
+    background: #006eaa;
+    box-shadow: 0 0 0 1px transparent inset;
+	cursor: pointer;
+    /* display: inline-block; */
```
Comment (gitjul) [inline client/public/home.css:385]: we shouldn't keep commented code as a rule

File: client/public/home.css [hunk 13]
```diff
@@ -0,0 +1,402 @@
...
+    padding: .78571429em 1.5em .78571429em;
+    text-transform: none;
+    text-shadow: none;
+	line-height: 1em;
+    font-style: normal;
+    text-align: center;
+    text-decoration: none;
+    border-radius: .28571429rem;
+	/* grid-column-start:2;
+	grid-column-end: end; */
```
Comment (gitjul) [inline client/public/home.css:401]: we shouldn't keep commented code as a rule

File: client/src/AuthProvider.js [hunk 14]
```diff
@@ -27,8 +27,34 @@ export default (C) =>
...
+      
+    }
+
+    async tokenReceivedCallback(response){
+      console.log(response);
+    }
+
+    async errorReceivedCallback(response){
+      console.log(response);
+    }
```
Comment (gitjul) [inline client/src/AuthProvider.js:40]: is there a plan for those? or were you just using it for development?

File: api/routes/userList.js [hunk 15]
```diff
@@ -39,23 +39,29 @@ class UserList{
...
+            user.country = verifiedUser.country;
+            user.profilelink = verifiedUser.profilelink;
+            user.anonymous = verifiedUser.anonymous;
+            user.profilestatus = verifiedUser.profilestatus;
+            user.email = verifiedUser.email;
+            await this.userDao.updatesignintime(user);
+        }
+        res.send(user);
+      }catch(error){
+        console.log(error);
```
Comment (aviskarkc10) [inline api/routes/userList.js:61]: let's remove console

File: client/src/AuthProvider.js [hunk 16]
```diff
@@ -54,9 +80,11 @@ export default (C) =>
 
     async onSignIn(redirect) {
       if (redirect) {
-        return msalApp.loginRedirect(GRAPH_REQUESTS.LOGIN);
+        return msalApp.loginRedirect({
+              scopes: GRAPH_REQUESTS.LOGIN.scopes,
+              redirectUri: "http://localhost:3000"
```
Comment (aviskarkc10) [inline client/src/AuthProvider.js:85]: i think this shouldn't be here?

**Comment by mrememisaac** [comment]
> thanks @mrememisaac! great job!
> I left some comments, I haven't looked at the functionality in the browser at all yet but I want to do it tomorrow.
> 💪

Thanks. Will resolve

File: client/public/home.css (hunk 1 above)
Comment (mrememisaac) [reply client/public/home.css:6]: This file was not edited with an IDE. Was faster to use Azures editor for real time testing. Will fix

File: client/src/AuthProvider.js (hunk 14 above)
Comment (mrememisaac) [reply client/src/AuthProvider.js:40]: Its gone

File: client/src/AuthProvider.js (hunk 16 above)
Comment (mrememisaac) [reply client/src/AuthProvider.js:85]: Replaced with env var

**Comment by sethuraj** [comment]
merging this to test and review the changes

File: client/src/auth-utils.js [hunk 17]
```diff
@@ -23,6 +23,7 @@ export const fetchMsGraph = async (url, accessToken) => {
 };
 
 export const isIE = () => {
+  return true; //this ensures we are always redirecting, never popping up
   const ua = window.navigator.userAgent;
```
Comment (gitjul) [inline client/src/auth-utils.js:27]: 1. all the following code is now useless and can be removed
2. the function isn't even needed, why not just set true where it's called

**Pull request merged**